lib/stoplight/data_store/redis.rb

```diff
@@ -76,6 +79,15 @@ def clear_state(light)
         normalize_state(state)
       end
 
+      def notification_lock_exists?(light)
+        lock, = @redis.multi do |transaction|
+          transaction.exists?(notification_lock_key(light))
+          transaction.setex(notification_lock_key(light), @lock_ttl, LOCKED_STATUS)
+        end
```

Comment:
@Lokideos can we use [SETNX](https://redis.io/commands/setnx/) here?

Reply:
@bolshakov if we use `setnx` here we also have to clear the state in Redis to remove this key since `setnx` does not perform the cleanup after set amount of time.
I will think about a way to implement it, though.